### ISSUE TITLE
iDOS2 Bluetooth Keyboard

### ISSUE THREAD

**Issue by convolutionbbs** [litchie/dospad#9]
I recently installed iDOS2 off the app store and was able to get some SCI games running well. I have an iPad3/4 with a Belkin external Bluetooth keyboard. The numbers, letters, and arrow keys work - but the tab and command (ctrl) do not. 

Is this a limitation in the port or is there a way to configure this?  It may be necessary to find out what key dos thought I was pressing when I typed the tab key would be needed.

Thanks!

**Comment by litchie** [comment]
Fully functional keyboard requires a developer build or jailbreak.

**Comment by jguice** [comment]
On my keyboard the arrow keys don't work consistently (or sometimes at all).  It seems like this should be fixable within the confines of a non-jailbroken app given that all the keys work in many other apps.

Perhaps it requires a small listener that intercepts keypresses in the usual ObjC way and "replays" them to DosBox?

I should mentioned I'm running the app store version of iDOS 2.  Does this source represent that version?  If so I'd be happy to look at improving keyboard support. :)

**Comment by litchie** [comment]
yes, this is the source. If you build the dospad target, you will have a fully functional keyboard. Appstore version (iDOS) can't use private API, so that's why we have limited keyboard support in iDOS2.

**Comment by jguice** [comment]
Cool, I'll take a look.  Thanks for making this.  It's really awesome! :smile_cat:

**Comment by jguice** [comment]
I'm not finding the app store version here in the code.  The iDOS app appears to be the 1.x line (with the pc-style custom keyboard) and of course DosPad is the Cydia version.  Is there a flag/scheme I'm missing? :)

**Comment by litchie** [comment]
v2.1  appstore version has some improvements regarding bluetooth keyboard.  
Now CTRL-[A-Z] and ALT-[A-Z] are supported. function keys are emulated with COMMAND-0, COMMAND-1, ..., and so on. 
That's about all we can do at this point.
